cpp/farm_ng/core/prototools/event_log_reader.cpp

```diff
@@ -158,7 +158,9 @@ bool EventTimeCompare::operator()(
   auto maybe_lhs_stamp = getStamp(lhs.event(), clock_name, semantics);
   auto maybe_rhs_stamp = getStamp(rhs.event(), clock_name, semantics);
   if (maybe_lhs_stamp && maybe_rhs_stamp) {
-    return maybe_lhs_stamp->stamp() < maybe_rhs_stamp->stamp();
+    return maybe_lhs_stamp->stamp() < maybe_rhs_stamp->stamp() ||
+           (!(maybe_rhs_stamp->stamp() < maybe_lhs_stamp->stamp()) &&
+            lhs.event().uri().path() < rhs.event().uri().path());
   }
   return false;
```

Comment:
@ethanrublee  Just noticed the false here.

This means that the operator does not satisfy **Strict Weak Ordering** because it violates:
"f(x, y) implies !f(y, x)" https://www.boost.org/sgi/stl/StrictWeakOrdering.html

Using this comparison operator in functions like std::sort: https://medium.com/@shiansu/strict-weak-ordering-and-the-c-stl-f7dcfa4d4e07

Since UB is strictly worse than a crash I would suggest either:

1. To crash when the stamp is not available or similar. 
2. Rename the function "fooBar()" and document that it is not strictly weak ordered.

Reply:
Yes this should give an error.

Reply:
Ok make this CHECK fail if this pre-condition is not met that both events contain the stamp and the semantics string.